src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -347,6 +361,54 @@ export const moveEncryptedPayload = async (
   return next()
 }
 
+/**
+ * Moves encrypted payload present in req.body to req.formsg.encryptedPayload.
+ * Should only be used for the old storage mode submission endpoint (/api/v3/forms/:formId/submissions/encrypt).
+ */
+export const validateEncryptSubmission = async (
+  req: EncryptSubmissionMiddlewareHandlerRequest,
+  res: Parameters<EncryptSubmissionMiddlewareHandlerType>[1],
```

Comment:
It took me a while to understand this comment 😅 

> Okay so after investigation, it's because the IncomingEncryptSubmission.init also validates the fields again, which in the case of the old endpoint, will just be for the mobile and email fields. This validation is no longer required for the new endpoint as it can simply share what email mode is using.
Since the softening of storage submission validation will not work without the fix, the fix is included here. This solves the validator bug seen when we attempted to rollout too. I'll update the description and tests in a bit.

Would be useful to add in the documentation here that:
- This middleware also validates mobile and email field via `IncomingEncryptSubmission.init`
- This is applied only for old storage mode submission endpoint (...) because 
- This applies only to old e2ee encrypt mode submission because storage mode submissions has another middleware `validateStorageSubmission` which handles validation (on that note, do add docs for `validateStorageSubmission` explaining that it is the same validation as for email mode!

Reply:
Ah, so sorry about the comment! It was copied over from the `moveEncryptedPayload` middleware and I forgot to change it.

Updated in [511413d](https://github.com/opengovsg/FormSG/pull/6696/commits/511413db4d9aadda5d229122c869ed5dbf995355)!